asf_search/ASFSearchOptions/validator_map.py

```diff
@@ -2,8 +2,8 @@
 
 from .validators import (
     parse_string, parse_float, parse_wkt, parse_date,
-    parse_string_list, parse_int_list, parse_int_or_range_list,
-    parse_float_or_range_list,
+    parse_string_iterator, parse_int_iterator, parse_int_or_range_iterator,
+    parse_float_or_range_iterator,
```

Comment:
Out of curiosity, is `iterator` more of a standard? I see cdk use it, only because their docs are for multiple programming languages. Imo `list` is the python equivalent, and more readable for python users.

Reply:
Also, does `parse_string_iterator/list` still behaves as expected? Str might be itself in itr, so you might get back something like `["m", "y", "_", "s", "t", "r" ]` instead of `["my_str"]`

Reply:
I think if it saves users the trouble of creating a list it's worth it, and given that we've labeled `search()` methods as accepting iterables it's worth actually supporting iterables.

And parsing strings is a gotcha, which is why `parse_iterator()` now checks if the value is a string and wraps it in a list.
https://github.com/asfadmin/Discovery-asf_search/pull/252/files/43541fd294656a3b139f8d6eb9bf268704b029f5#diff-46defa23f95a69e0968b875aafa178f9e30730631a56e98e06c20ec16dc2ba20R98

Reply:
Ah I missed the string catch, nice. There's other corner cases like dicts too though that pass the iterator check, but we don't want. By having the hint being list, we're sort-of saying any list-like object. They can still pass in whatever they want and python will "try" regardless.

Definitely feel free to get other opinions on this too! That's just my initial hunch. The scope of iterator is just so large. Arguably instead of list, we could change it to "list and tuple" to make it extra clear what we want too.

Reply:
I just remembered AWS uses "Sequence" from typing, not Itterator. I like it more than ittr, but idk if it's better than saying list/tupple. Need to think/research it more

Reply:
Sequence seems like the better choice, it looks like it covers our edge case for dictionaries!